search_similar_entities/facetedSearchEndpoint.rb

```diff
@@ -0,0 +1,161 @@
+#QuerySimilar Endpoint
+
+#Builds complex query for"Query Similar" functionality; returns a link to the search page, with a preseeded query looking for all entities of type testEntity that share certain attributes with the current "testEntity" entity, or alternatively, the same query but with preselected facets
+
+# Verify params are OK
+raise "Missing required parameter: id" unless params[:id]
```

Comment:
I used the "Using parameters" example code:

```
# Using parameters
# Accesses the values that are passed to the endpoint with the <params> array
name = params[:parameter_name]

# Checks for a required parameter, and raises an exception if the parameter is not supplied
raise "Missing required parameter" unless params[:required_parameter_name]

# Uses a parameter value if one is passed to the endpoint, and a default value if one is not passed to the endpoint
message = params[:message] || "default message"

# Converts a parameter to an integer, possibly to do computations with the value
count = params[:some_number].to_i
#

```
If I replace "raise" by return, I noticed that the Test Endpoint functionality never shows the returned message